### PR TITLE
Update item templates

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
Updating to reflect changes in the oM with regards to the attributes.

Also, making use of the `$year$` variable to avoid us having to update the templates every year.

While at it, also added a template for adding a new BHoMObejct

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
Missed that the templates had been modified before by @alelom .

I personally prefer to keep the templates clean without extra "how to"s, but happy to be overruled and add the comments back in.

File: templates/Engine method templates/BHoMObjectTemplate/MyTemplate.vstemplate [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <Name>BHoMObject - Add a new class</Name>
+    <Description>Adds a BHoMObject to your oM project.</Description>
+    <ProjectType>CSharp</ProjectType>
+    <SortOrder>10</SortOrder>
+    <Icon>CreateBHoM.ico</Icon>
+  </TemplateData>
+  <TemplateContent>
+    <References>
+      <Reference>
+        <Assembly>BHoM</Assembly>
```
Comment (FraserGreenroyd) [inline templates/Engine method templates/BHoMObjectTemplate/MyTemplate.vstemplate:13]: Can we not hardcode the hint path to the `C:/ProgramData/BHoM/Assemblies` folder or is that not possible?

File: templates/Engine method templates/ConvertTemplate/ConvertTemplate.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+namespace $rootnamespace$ //`Convert` is a partial class. Remove any reference to `Convert` the from namespace.
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("")]
+		[Input("","")]
```
Comment (FraserGreenroyd) [inline templates/Engine method templates/ConvertTemplate/ConvertTemplate.cs:39]:
```suggestion
	[Input("","")]
```

File: templates/Engine method templates/ConvertTemplate/ConvertTemplate.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        [Output("", "")]
+        public static void $safeitemname$()
+        {
+			throw new NotImplementedException();
```
Comment (FraserGreenroyd) [inline templates/Engine method templates/ConvertTemplate/ConvertTemplate.cs:43]:
```suggestion
		throw new NotImplementedException();
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
One question, couple of formatting changes.

I'm happy without the extra comments, I think it makes sense to just use the wiki as a single source of documentation so we only ever had to update things once.

File: templates/Engine method templates/BHoMObjectTemplate/MyTemplate.vstemplate (hunk 1 above)
Comment (IsakNaslundBh) [reply templates/Engine method templates/BHoMObjectTemplate/MyTemplate.vstemplate:13]: This is for an item, not a project. From memory, this is just stating "This item template requires BHoM to be used by this project"

Comment (FraserGreenroyd) [reply templates/Engine method templates/BHoMObjectTemplate/MyTemplate.vstemplate:13]: Ok, am happy

**Review by FraserGreenroyd — APPROVED** [review]
All LGTM now on review, tack @IsakNaslundBh

File: templates/Engine method templates/ComputeTemplate/ComputeTemplate.cs [hunk 4]
```diff
@@ -24,10 +24,10 @@
 using System.Linq;
 using System.Collections.Generic;
 using System.ComponentModel;
-using BH.oM.Reflection.Attributes;
+using BH.oM.Base.Attributes;
 using BH.oM.Base;
 
-namespace $rootnamespace$ //Compute is a partial class. Remove any reference to Compute from the namespace
+namespace $rootnamespace$ //`Compute` is a partial class. Remove any reference to `Compute` the from namespace.
```
Comment (alelom) [inline templates/Engine method templates/ComputeTemplate/ComputeTemplate.cs:30]: Introduced typo `the from namespace` - sorry I didn't review in time.

Comment (alelom) [reply templates/Engine method templates/ComputeTemplate/ComputeTemplate.cs:30]: Typo is repeated in every file

**Pull request merged**